src/cloudsc_cuda/cloudsc/load_state.cu

```diff
@@ -132,57 +140,79 @@ void expand_3d(double *buffer_in, double *field_in, int nlon, int nlev, int nclv
 
 #ifdef HAVE_SERIALBOX
 void load_and_expand_1d(serialboxSerializer_t *serializer, serialboxSavepoint_t* savepoint,
-    const char *name, int nlon, int nproma, int ngptot, int nblocks, double *field)
+    const char *name, int nlon, int nproma, int ngptot, int nblocks, dtype *field)
 {
-  double buffer[nlon];
+  double dbl_buffer[nlon];
+  dtype buffer[nlon];
   int strides[1] = {1};
 
-  serialboxSerializerRead(serializer, name, savepoint, buffer, strides, 1);
-  expand_1d((double *)buffer, field, nlon, nproma, ngptot, nblocks);
+  serialboxSerializerRead(serializer, name, savepoint, dbl_buffer, strides, 1);
+  for (int i=0;i<nlon;i++) {
+    buffer[i] = (dtype) dbl_buffer[i];
+  }
+  expand_1d((dtype *)buffer, field, nlon, nproma, ngptot, nblocks);
 }
 
 void load_and_expand_1d_int(serialboxSerializer_t *serializer, serialboxSavepoint_t* savepoint,
     const char *name, int nlon, int nproma, int ngptot, int nblocks, int *field)
 {
-  int buffer[nlon];
+  double buffer[nlon];
```

Comment:
This should load ints, why does the buffer have to be double?

Reply:
there is no reason ... not sure why I changed that